src/models/FormEditMetadata.ts

```diff
@@ -5,14 +5,24 @@ export const validationSchema = Yup.object().shape({
   name: Yup.string()
     .min(4, (param) => `Title must be at least ${param.min} characters`)
     .required('Required'),
-  description: Yup.string().required('Required').min(10)
+  description: Yup.string().required('Required').min(10),
+  timeout: Yup.string().required('Required')
 })
 
 export function getInitialValues(
-  metadata: MetadataMarket
+  metadata: MetadataMarket,
+  timeout: number
 ): Partial<MetadataPublishForm> {
   return {
     name: metadata.main.name,
-    description: metadata.additionalInformation?.description
+    description: metadata.additionalInformation.description,
+    timeout:
+      timeout === 0
```

Comment:
while it's fine hardcoding our own mapping in the publish form, we can't do that for any display of existing timeout in the DDO. Background is that users can push into market without using the UI, with libraries or other interfaces and this always needs to be considered. Most of the time we will encounter `timeout` values we have defined in publish form, but if not, the value display can't be broken.

So I think we do not get around using some library for this, maybe only overwriting its behavior for mapping `0` to `Forever`. This one was mentioned somewhere else https://github.com/vercel/ms but it seem to only deal with milliseconds. Unifying this mapping will also be important in combination with https://github.com/oceanprotocol/market/pull/317 where we display timeout in more places.

Reply:
I'm not sure we can have a custom option, to actually show the exact value. Maybe we can calculate and add it somewhere in the form and alter `initialValues` and `content.form`. While I agree with you I'm not sure we need to focus that much at this scenario at this point.

Or the easiest way is to put a simple input in edit, not select. And then the user just adds the number of seconds. This way it becomes a bad UX because the user needs to calculate how many seconds are in 1 week or 1 month etc.

Ideally we have a control that adds predefined values in the input, but it implies more dev time. Similar to a date picker, you can manually add the number of seconds but you can also select predetermined values

Reply:
manually entering seconds indeed is a bit weird. Think then the compromise is simply having some fallback for the case of a timeout not within our presets, this change here would always output `1 year` in that case. It should then just output the existing value from DDO, but feels like we can simply do the conversion for user in this case too. How about simply converting from seconds to hours for the fallback as a compromise? So we don't introduce new library for like 0.5% of use cases, and it's more human-friendly than seconds